### PR TITLE
Small fixes for compatibility with the Perseids Treebank Template

### PR DESCRIPTION
**Pull request by zfletch**
* Remove `resize` functionality because it results in an infinitely resizing canvas when the user changes the page width.
* Add `getSubdoc` to the API to support https://github.com/perseids-publications/treebank-template/pull/11
* Add d3 to deploy script

### CODE REVIEW

File: app/js/arethusa.core/services/navigator.js [hunk 1]
```diff
@@ -95,6 +95,9 @@ angular.module('arethusa.core').service('navigator', [
     function currentIds () {
       return arethusaUtil.map(currentSentenceObjs(), 'id');
     }
+    this.currentSubdocs = function() {
+      return arethusaUtil.map(currentSentenceObjs(), 'subdoc');
```
Comment (zfletch) [inline app/js/arethusa.core/services/navigator.js]: Do you know if there's any reason not to return the first element of this array? In the treebanks I tested this with there was never more (or less) than one element in this array.

File: deploy_widget.sh [hunk 2]
```diff
@@ -4,12 +4,12 @@ then
...
 fi
 echo "Deploying to $deploy_dir"
-if [ ! -e $deploy_dir ]
-then
-  mkdir $deploy_dir
-  mkdir $deploy_dir/dist
-fi
+
+mkdir -p $deploy_dir/dist
+mkdir -p $deploy_dir/../vendor/d3-3.4.13/ # TODO this doesn't seem right but that's the value in app/js/arethusa.dep_tree/services/dep_tree.js
```
Comment (zfletch) [inline deploy_widget.sh]: In `app/js/arethusa.dep_tree/services/dep_tree.js` the path to `d3` is `../../vendor/d3-3.4.13/d3.js`. Does that mean that when it is deployed it looks for d3 at that location?

File: deploy_widget.sh [hunk 3]
```diff
@@ -4,12 +4,12 @@ then
...
+
 cp dist/arethusa.min.js $deploy_dir
 cp dist/arethusa.min.map $deploy_dir
 cp dist/arethusa_packages.min.js $deploy_dir/arethusa.packages.min.js
 cp dist/* $deploy_dir/dist
+cp vendor/d3-3.4.13/d3.js $deploy_dir/../vendor/d3-3.4.13/d3.js # TODO should this not be d3.min.js?
```
Comment (zfletch) [inline deploy_widget.sh]: Related to the comment above, `dep_tree.js` is requiring `../../vendor/d3-3.4.13/d3.js`. Should that be changed to `d3.min.js`?

File: app/js/arethusa.core/services/navigator.js (hunk 1 above)
Comment (balmas) [reply app/js/arethusa.core/services/navigator.js]: The subdoc attribute is optional. It is always used on treebank files that are created through the Perseids workflow (I think), but it's possible that someone could upload a treebank file that doesn't use that field. But in that case you would just get an empty array.  There should never been more than one subdoc per sentence using the current treebank schema.

File: deploy_widget.sh (hunk 2 above)
Comment (balmas) [reply deploy_widget.sh]: I struggled with getting this path right.  I was trying to match what you had in the treebank-template with what Arethusa expected. The only way I could get it work was to use the relative path in the arethusa dep_tree.js file.

File: deploy_widget.sh (hunk 3 above)
Comment (balmas) [reply deploy_widget.sh]: We probably should use the minified file. As with above, I was just trying to make arethusa match what was in the treebank template.

File: app/js/arethusa.core/directives/fullHeight.js [hunk 4]
```diff
@@ -17,9 +17,6 @@ angular.module('arethusa.core').directive('fullHeight', [
           var fullHeight = body.height();
           element.height(fullHeight - margin - padding - additionalBorder);
         }
-        win.bind('resize', function() {
-          resize();
-        });
```
Comment (balmas) [inline app/js/arethusa.core/directives/fullHeight.js]: so, without this, the tree stays the same size when the window resizes?  We might want to add an API call to allow embedding code to resize the tree then.

Comment (zfletch) [reply app/js/arethusa.core/directives/fullHeight.js]: Interestingly, the tree still resizes without this code. With this code, resizing the screen horizontally causes the tree to grow vertically. That meant that every time the user resized the screen the tree would grow longer and longer.

File: app/js/arethusa.core/services/navigator.js (hunk 1 above)
Comment (zfletch) [reply app/js/arethusa.core/services/navigator.js]: Great, I'll update this to `arethusaUtil.map(currentSentenceObjs(), 'subdoc')[0]`. That should make the API easier.

File: deploy_widget.sh (hunk 2 above)
Comment (zfletch) [reply deploy_widget.sh]: Do you think it makes more sense to be in `$deploy_dir/../vendor/` or `$deploy_dir/vendor`? We can change the Treebank Template to use whatever path makes the most sense here.

File: app/js/arethusa.core/directives/fullHeight.js (hunk 4 above)
Comment (balmas) [reply app/js/arethusa.core/directives/fullHeight.js]: ok, thanks. I think it's fine, and if it turns out we need to add something to the api, we can.

File: deploy_widget.sh (hunk 2 above)
Comment (balmas) [reply deploy_widget.sh]: I have no idea :-)

Comment (zfletch) [reply deploy_widget.sh]: I'll keep it as is then!

**Pull request merged**